### PR TITLE
Release/v4.3.7

### PR DESCRIPTION
**Pull request by michaelcfanning**
## **v4.3.7** [Sdk](https://www.nuget.org/packages/Sarif.Sdk/v4.3.7) | [Driver](https://www.nuget.org/packages/Sarif.Driver/v4.3.7) | [Converters](https://www.nuget.org/packages/Sarif.Converters/v4.3.7)  | [Multitool](https://www.nuget.org/packages/Sarif.Multitool/v4.3.7) | [Multitool Library](https://www.nuget.org/packages/Sarif.Multitool.Library/v4.3.)
* DEP: Updated NewtonSoft.JSON to 8.0.3 in Sarif.Converters for .NET targets later than `netstandard2.0`.
* BUG: Logging improved when work item client is called with invalid work item values.
* NEW: Add `Path.Combine`, `Path.GetDirectoryName` and `Path.GetFileNameWithoutExtension` to `IFileSystem`.

### CODE REVIEW

**Review by scottoneil-ms — APPROVED** [review]

File: src/Sarif/IFileSystem.cs [hunk 1]
```diff
@@ -330,5 +319,49 @@ public interface IFileSystem
...
         FileVersionInfo FileVersionInfoGetVersionInfo(string fileName);
+
+        /// <summary>
+        /// Combines an array of strings into a path.
+        /// </summary>
+        /// <param name="paths">
+        /// An array of parts of the path.
+        /// </param>
+        /// <returns>
+        /// The combined paths.
```
Comment (yongyan-gh) [inline src/Sarif/IFileSystem.cs:330]: >paths [](http://example.com/codeflow?start=25&length=5)

nit: path

**Review by yongyan-gh — APPROVED** [review]
:shipit:

File: src/Sarif/FileSystem.cs [hunk 2]
```diff
@@ -413,5 +399,61 @@ public FileVersionInfo FileVersionInfoGetVersionInfo(string fileName)
...
+        /// Returns the directory information for the specified path.
+        /// </summary>
+        /// <param name="path">
+        /// The path of a file or directory.
+        /// </param>
+        /// <returns>
+        /// Directory information for path, or null if path denotes a root directory or is null. Returns <see cref="string.Empty"/> if path does not contain directory information.
+        /// </returns>
+        public string PathGetDirectoryName(string path)
+        {
```
Comment (yongyan-gh) [inline src/Sarif/FileSystem.cs:427]: >  [](http://example.com/codeflow?start=9&length=1)

couple dotnet format issues

\FileSystem.cs(427,10): error WHITESPACE: Fix whitespace formatting. Replace 15 characters with '\r\n\s\s\s\s\s\s\s\s\s\s\s\s'. [D:\a\sarif-sdk\sarif-sdk]
\FileSystem.cs(428,48): error WHITESPACE: Fix whitespace formatting. Replace 11 characters with '\r\n\s\s\s\s\s\s\s\s'. [D:\a\sarif-sdk\sarif-sdk]
\FileSystem.cs(454,67): error WHITESPACE: Fix whitespace formatting. Replace 11 characters with '\r\n\s\s\s\s\s\s\s\s'. [D:\a\sarif-sdk\sarif-sdk]

File: src/Sarif/IFileSystem.cs (hunk 1 above)
Comment (michaelcfanning) [reply src/Sarif/IFileSystem.cs]: Love it, nice catch, sir.

**Pull request merged**